src/key.cpp

```diff
@@ -128,16 +147,21 @@ AsymmetricKey ECCSpec::generate() const {
 
 std::unique_ptr<AsymmetricKey::Spec> AsymmetricKey::getKeySpec() const
 {
-    if(getType() == KeyTypes::ECC) {
+    if (getType() == KeyTypes::ECC){
+        auto nid = openssl::_EVP_PKEY_type(_key.get());
         openssl::ellipticCurveNid keyNid;
-        try {
-            const EC_GROUP *group = _EC_KEY_get0_group(_EVP_PKEY_get0_EC_KEY(_key.get()));
-            keyNid = static_cast<openssl::ellipticCurveNid>(_EC_GROUP_get_curve_name(group));
-        } catch (const OpenSSLException &e) {
-            throw MoCOCrWException(e.what());
+        if(nid == EVP_PKEY_EC) {
+            try {
+                const EC_GROUP *group = _EC_KEY_get0_group(_EVP_PKEY_get0_EC_KEY(_key.get()));
+                keyNid = static_cast<openssl::ellipticCurveNid>(_EC_GROUP_get_curve_name(group));
+            } catch (const OpenSSLException &e) {
+                throw MoCOCrWException(e.what());
+            }
+        } else {
```

Comment:
A generic else looks potentially dangerous to me. Please check exactly what we expect here and throw if something unexpected arrives here.